testing/conftest.py

```diff
@@ -136,6 +136,23 @@ def pytester(pytester: Pytester, monkeypatch: MonkeyPatch) -> Pytester:
     return pytester
 
 
+class _ColorMapping:
+    @classmethod
+    def format(cls, lines: List[str]) -> List[str]:
+        """Straightforward replacement of color names to their ASCII codes."""
+        return [line.format(**cls.COLORS) for line in lines]
+
+    @classmethod
+    def format_for_fnmatch(cls, lines: List[str]) -> List[str]:
+        """Replace color names for use with LineMatcher.fnmatch_lines"""
+        return [line.format(**cls.COLORS).replace("[", "[[]") for line in lines]
+
+    @classmethod
+    def format_for_rematch(cls, lines: List[str]) -> List[str]:
+        """Replace color names for use with LineMatcher.re_match_lines"""
+        return [line.format(**cls.RE_COLORS) for line in lines]
```

Comment:
Might want to create `RE_COLORS` here (or in an `__init__`, but I think it's fine to do it every time the function is called). That way, you don't need to duplicate `RE_COLORS = {k: re.escape(v) for k, v in COLORS.items()}` for all subclasses, and mypy will be happy as well.